cmd/cli/plugin/airgapped-network/imageop/publishimagestotar.go

```diff
@@ -0,0 +1,269 @@
+	}
+	tkgBomImagePath := path.Join(p.TkgImageRepo, "tkg-bom")
+
+	sourceImageName := tkgBomImagePath + ":" + p.TkgVersion
+	tarnames := "tkg-bom" + "-" + p.TkgVersion + ".tar"
+	destRepo := path.Join(p.DestinationRepository, tkgBomImagePath)
+	p.ImageDetails[tarnames] = destRepo
+	err := p.PkgClient.CopyImageToTar(sourceImageName, tarnames)
+	if err != nil {
+		return "", errors.New("error while downloading tkg-bom")
+	}
+	err = p.PkgClient.PullImage(sourceImageName, outputDir)
+	if err != nil {
+		return "", err
+	}
+	// read the tkg-bom file
+	tkgBomFilePath := path.Join(outputDir, fmt.Sprintf("tkg-bom-%s.yaml", p.TkgVersion))
+	b, err := os.ReadFile(tkgBomFilePath)
+
+	// read the tkg-bom file
+	if err != nil {
+		return "", errors.Wrapf(err, "read tkg-bom file from %s faild", tkgBomFilePath)
+	}
+	tkgBom, _ := tkrv1.NewBom(b)
+	// imgpkg copy each component's artifacts
+	components, _ := tkgBom.Components()
+	group, _ := errgroup.WithContext(context.Background())
+	for _, compInfos := range components {
+		for _, compInfo := range compInfos {
+			for _, imageInfo := range compInfo.Images {
+				sourceImageName = path.Join(p.TkgImageRepo, imageInfo.ImagePath) + ":" + imageInfo.Tag
+				destImageRepo := path.Join(p.DestinationRepository, imageInfo.ImagePath)
+				imageInfo.ImagePath = replaceSlash(imageInfo.ImagePath)
+				tarname := imageInfo.ImagePath + "-" + imageInfo.Tag + ".tar"
+				p.ImageDetails[tarname] = destImageRepo
+				group.Go(func() error {
+					return p.PkgClient.CopyImageToTar(sourceImageName, tarname)
+				})
+			}
+		}
+	}
+	err = group.Wait()
+	if err != nil {
+		return "", errors.Wrap(err, "error while downloading images")
+	}
+
+	return tkgBom.GetCompatibility(), nil
+}
+
+func (p *PublishImagesToTarOptions) DownloadTkrCompatibilityImage(tkrCompatibilityRelativeImagePath string) (tkgVersion []string, err error) {
+	if p.TkgImageRepo == "" || p.TkgVersion == "" {
+		return nil, errors.New("input parameter TkgImageRepo or TkgVersion is empty")
+	}
+
+	// get the latest tag of tkr-compatibility image
+	tkrCompatibilityImagePath := path.Join(p.TkgImageRepo, tkrCompatibilityRelativeImagePath)
+	imageTags := p.PkgClient.GetImageTagList(tkrCompatibilityImagePath)
+	if len(imageTags) == 0 {
+		return nil, errors.New("image doesn't have any tags")
+	}
+	// inspect the tkr-compatibility image to get the list of compatible tkrs
+	tkrCompatibilityImageURL := tkrCompatibilityImagePath + ":" + imageTags[len(imageTags)-1]
+
+	sourceImageName := tkrCompatibilityImageURL
+	err1 := p.PkgClient.PullImage(sourceImageName, outputDir)
+	if err1 != nil {
+		return nil, err1
+	}
+	files, err := os.ReadDir(outputDir)
+	if err != nil {
+		return nil, errors.Wrapf(err, "read directory tmp failed")
+	}
+	if len(files) != 1 || files[0].IsDir() {
+		return nil, fmt.Errorf("tkr-compatibility image should only has exact one file inside")
+	}
+	tkrCompatibilityFilePath := path.Join(outputDir, files[0].Name())
```

Comment:
```suggestion
	tkrCompatibilityFilePath := filepath.Join(outputDir, files[0].Name())
```

Reply:
Hi @joaopapereira  , Please check your below comment , i didn't test it on window so i wasn't sure the behavior hence I made the changes as per your suggestion in below comment
https://github.com/vmware-tanzu/tanzu-framework/pull/3743#discussion_r1028538876